src/Domain/Creators/TimeDependentOptions/Sphere.cpp

```diff
@@ -183,24 +185,79 @@ TimeDependentMapOptions::create_functions_of_time(
 }
 
 void TimeDependentMapOptions::build_maps(
-    const std::array<double, 3>& center,
-    std::pair<double, double> inner_shell_radii,
-    std::pair<double, double> outer_shell_radii) {
+    const std::array<double, 3>& center, const bool filled,
+    const double inner_radius, const std::vector<double>& radial_partitions,
+    const double outer_radius) {
+  filled_ = filled;
   if (shape_map_options_.has_value()) {
     std::unique_ptr<domain::CoordinateMaps::ShapeMapTransitionFunctions::
                         ShapeMapTransitionFunction>
-        transition_func =
-            std::make_unique<domain::CoordinateMaps::
-                                 ShapeMapTransitionFunctions::SphereTransition>(
-                inner_shell_radii.first, inner_shell_radii.second);
-    shape_map_ = ShapeMap{center,
-                          shape_map_options_->l_max,
-                          shape_map_options_->l_max,
-                          std::move(transition_func),
-                          shape_name,
-                          size_name};
+        transition_func;
+    if (filled_) {
+      if (transition_rot_scale_trans_ and radial_partitions.size() < 2) {
+        ERROR(
+            "Currently at least two radial partitions are required to "
+            "transition the RotScaleTrans map to zero in the outer shell "
+            "when a shape map is present.");
+      }
+      using WedgeTransition =
+          domain::CoordinateMaps::ShapeMapTransitionFunctions::Wedge;
+      // Shape map transitions from 0 to 1 from the inner cube to this surface
+      deformed_radius_ =
+          radial_partitions.empty() ? outer_radius : radial_partitions.front();
+      // Shape map transitions from 1 to 0 from the deformed surface to the next
+      // radial partition or to the outer radius
+      const bool has_shape_rolloff = not radial_partitions.empty();
+      const double shape_outer_radius =
+          radial_partitions.size() > 1 ? radial_partitions[1] : outer_radius;
```

Comment:
[optional] Combine these into a `std::optional<double>`

Reply:
Not sure what you think could be combined into an optional

Reply:
```cpp
const std::optional<double> shape_outer_radius = 
    radial_partitions.empty() ? std::nullopt : 
    (radial_partitions.size() > 1 ? radial_partitions[1] : outer_radius);


for (size_t j=0; j < (shape_outer_radius.has_value() ? 12 : 6); ++j) {
  if (j < 6) {
    ...
  } else {
    // use shape_outer_radius.value()
  }
}
```

Reply:
Oh I see now, thanks for explaining. I don't think that makes the code much cleaner or more readable, so I'll just keep it as is